### PR TITLE
[Synapse] Added support for long running operation and added some generated model classes in Artifacts

### PR DESCRIPTION
**Pull request by wonner**
Swagger PR: https://github.com/Azure/azure-rest-api-specs/pull/10270

### CODE REVIEW

**Review by idear1203 — APPROVED** [review]

File: sdk/synapse/Azure.Analytics.Synapse.Artifacts/CHANGELOG.md [hunk 1]
```diff
@@ -1,4 +1,10 @@
 # Release History
 
-## 1.0.0-preview.1 (Unreleased)
+## 1.0.0-preview.2 (Unreleased)
+
+- Added support for long running operation.
+
+- Added generated model classes.
+
+## 1.0.0-preview.1
```
Comment (tg-msft) [inline sdk/synapse/Azure.Analytics.Synapse.Artifacts/CHANGELOG.md:9]: Nit - you should add the date of your last preview like `## 1.0.0-preview.1 (2020-06-31)` or whenever the actual date was.  That's the format we're using for all these.

File: sdk/synapse/Azure.Analytics.Synapse.Artifacts/api/Azure.Analytics.Synapse.Artifacts.netstandard2.0.cs [hunk 2]
```diff
@@ -83,14 +83,40 @@ public partial class LinkedServiceClient
...
+        public virtual System.Threading.Tasks.Task<Azure.Analytics.Synapse.Artifacts.LinkedServiceCreateOrUpdateLinkedServiceOperation> StartCreateOrUpdateLinkedServiceAsync(string linkedServiceName, Azure.Analytics.Synapse.Artifacts.Models.LinkedServiceResource linkedService, string ifMatch = null, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        public virtual Azure.Analytics.Synapse.Artifacts.LinkedServiceDeleteLinkedServiceOperation StartDeleteLinkedService(string linkedServiceName, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        public virtual System.Threading.Tasks.Task<Azure.Analytics.Synapse.Artifacts.LinkedServiceDeleteLinkedServiceOperation> StartDeleteLinkedServiceAsync(string linkedServiceName, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+    }
+    public partial class LinkedServiceCreateOrUpdateLinkedServiceOperation : Azure.Operation<Azure.Analytics.Synapse.Artifacts.Models.LinkedServiceResource>
```
Comment (tg-msft) [inline sdk/synapse/Azure.Analytics.Synapse.Artifacts/api/Azure.Analytics.Synapse.Artifacts.netstandard2.0.cs:95]: Is the code generator picking these names automatically?  I think `CreateOrUpdateLinkedServiceOperation` sounds better and is just as clear.

Comment (tg-msft) [reply sdk/synapse/Azure.Analytics.Synapse.Artifacts/api/Azure.Analytics.Synapse.Artifacts.netstandard2.0.cs:95]: (It's fine to check these in now, but I think we should either tweak the swagger if we can control this ourselves or talk with the codegen folks to see if there's a way to customize this class name.)

File: sdk/synapse/Azure.Analytics.Synapse.Artifacts/api/Azure.Analytics.Synapse.Artifacts.netstandard2.0.cs [hunk 3]
```diff
@@ -1441,6 +1968,26 @@ public partial class DatasetZipDeflateCompression : Azure.Analytics.Synapse.Arti
...
+    public enum DaysOfWeek
+    {
+        Sunday = 0,
+        Monday = 1,
+        Tuesday = 2,
+        Wednesday = 3,
+        Thursday = 4,
+        Friday = 5,
+        Saturday = 6,
+    }
```
Comment (tg-msft) [inline sdk/synapse/Azure.Analytics.Synapse.Artifacts/api/Azure.Analytics.Synapse.Artifacts.netstandard2.0.cs:1990]: Can we unify these in the swagger?  We should also look at customizing the codegen to use https://docs.microsoft.com/en-us/dotnet/api/system.dayofweek?view=netcore-3.1 if we can.

File: sdk/synapse/Azure.Analytics.Synapse.Artifacts/api/Azure.Analytics.Synapse.Artifacts.netstandard2.0.cs [hunk 4]
```diff
@@ -3313,6 +4578,22 @@ public partial class RestServiceLinkedService : Azure.Analytics.Synapse.Artifact
...
+        public object RequestBody { get { throw null; } set { } }
+        public object RequestInterval { get { throw null; } set { } }
+        public object RequestMethod { get { throw null; } set { } }
+    }
+    public partial class RetryPolicy
+    {
+        public RetryPolicy() { }
+        public object Count { get { throw null; } set { } }
+        public int? IntervalInSeconds { get { throw null; } set { } }
+    }
```
Comment (tg-msft) [inline sdk/synapse/Azure.Analytics.Synapse.Artifacts/api/Azure.Analytics.Synapse.Artifacts.netstandard2.0.cs:4596]: This is another one of those things like Pipeline that's super overloaded in the REST client space and we should think about whether there's a better name we can use to make more Synapse specific in the future.

**Review by tg-msft — APPROVED** [review]
The changes are good to merge.  I think you're now officially the largest Track 2 client API at this point.  :smile:

We should think about whether there's a way we can group some of these different data sources into a couple of sub-namespaces in the future that make it easier for customers to find the model types they're looking for.

File: sdk/synapse/Azure.Analytics.Synapse.Artifacts/api/Azure.Analytics.Synapse.Artifacts.netstandard2.0.cs (hunk 2 above)
Comment (idear1203) [reply sdk/synapse/Azure.Analytics.Synapse.Artifacts/api/Azure.Analytics.Synapse.Artifacts.netstandard2.0.cs:95]: Sure, I took a note on this.

File: sdk/synapse/Azure.Analytics.Synapse.Artifacts/api/Azure.Analytics.Synapse.Artifacts.netstandard2.0.cs (hunk 4 above)
Comment (idear1203) [reply sdk/synapse/Azure.Analytics.Synapse.Artifacts/api/Azure.Analytics.Synapse.Artifacts.netstandard2.0.cs:4596]: Sure, we will fix this in future

File: sdk/synapse/Azure.Analytics.Synapse.Artifacts/api/Azure.Analytics.Synapse.Artifacts.netstandard2.0.cs (hunk 3 above)
Comment (idear1203) [reply sdk/synapse/Azure.Analytics.Synapse.Artifacts/api/Azure.Analytics.Synapse.Artifacts.netstandard2.0.cs:1990]: Good catch. We will send PR and fix this soon.

Comment (idear1203) [reply sdk/synapse/Azure.Analytics.Synapse.Artifacts/api/Azure.Analytics.Synapse.Artifacts.netstandard2.0.cs:1990]: Created a PR to fix this issue: https://github.com/Azure/azure-rest-api-specs/pull/10449

File: sdk/synapse/Azure.Analytics.Synapse.Artifacts/CHANGELOG.md (hunk 1 above)
Comment (idear1203) [reply sdk/synapse/Azure.Analytics.Synapse.Artifacts/CHANGELOG.md]: Fixed.

**Pull request merged**